dd-java-agent/agent-profiling/profiling-controller/src/main/java/com/datadog/profiling/jfr/SelfType.java

```
@@ -0,0 +1,41 @@
+package com.datadog.profiling.jfr;
+
+import java.util.List;
+
+/**
+ * A place-holder for fields of the same type as they are defined in. Keeping the default values
+ * intentionally 'wrong' in order for things to break soon if this type is not replaced by the
+ * concrete type counterpart correctly.
+ */
+final class SelfType extends BaseType {
+  static final SelfType INSTANCE = new SelfType();
+
+  private SelfType() {
+    super(Long.MIN_VALUE, "", null, null);
+  }
+
+  @Override
+  public boolean isBuiltin() {
+    return false;
+  }
+
+  @Override
+  public List<TypedField> getFields() {
+    return null;
```

should return Collections.emptyList()
as all other implementation of BaseType expect to return non-null
see BuiltinType

---

Rather, all the methods should throw an `UnsupportedOperationException` as this type is only a placeholder and never should escape to be used directly.